pkg/metrics/metrics.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	storageConfigured = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "storage",
+		Name:      "configured",
+		Help:      "Indicates if registry has configured storage.",
+	})
+	storageBackendCreated = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "storage_backend",
+		Name:      "created",
+		Help:      "Number of times registry storage backend got created.",
+	})
+	degraded = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "status",
+		Name:      "degraded",
+		Help:      "Indicates if registry is degraded.",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(
```

Comment:
we highly recommend not to use the global metrics registry, as for example that's how we ended up having etcd and apiserver metrics in the kubernetes controller manager, which is very confusing

Reply:
Could you point me to an already existing implementation to guide me here?

Reply:
The cluster-monitoring-operator does this. It instantiates and registers the default collectors here: https://github.com/openshift/cluster-monitoring-operator/blob/9dbc613664ffc99f5a82d82a05cd145dedcb863c/cmd/operator/main.go#L114-L118

And then the registry is passed around wherever needed, for example the operator then registers its metrics and collectors here: https://github.com/openshift/cluster-monitoring-operator/blob/9dbc613664ffc99f5a82d82a05cd145dedcb863c/cmd/operator/main.go#L132

It's a good idea to pass the registry around in a similar way you would with a logger instance.

Reply:
This is an excellent idea. As the code is organised now(we are centralizing all metrics into one single package) it is not entirely necessary but as the code grows this most likely will come handy. I sent a commit that address this, ptal.